src/CBLDocument.cc

```diff
@@ -92,6 +92,16 @@ alloc_slice CBLDocument::getRevisionHistory() const {
 }
 
 
+#pragma mark - Utils:
+
+
+void CBLDocument::checkCollectionMatches(CBLCollection* _cbl_nullable myCol, CBLCollection *colParam) {
+    if (myCol && *myCol != *colParam) {
```

Comment:
Is it standard practice to assume operator precedence and to elide the clarifying parenthesis?
If it were me, I'd parenthesize the second expression.

Reply:
The precedence && is lower than != so this check is fine. It doesn't hurt to put the parenthesis though given that there are multiple operator used in this if. I can do that.